blockly-core/tests/modal_editors_test.js

```diff
@@ -120,6 +134,28 @@ function test_initializeFunctionEditor() {
   assertEquals(false, definitionBlock.shouldBeGrayedOut());
   assertEquals(false, definitionBlock.isDeletable());
   assertEquals(false, definitionBlock.isEditable());
+  assertNull(Blockly.functionEditor.deleteButton_);
+
+  cleanupFunctionEditor();
+  goog.dom.removeNode(container);
+}
+
+function test_functionEditor_deleteButton() {
+  var container = Blockly.Test.initializeBlockSpaceEditor();
+  initializeFunctionEditor(USER_CREATED_PROCEDURE);
+  openFunctionEditor('test-usercreated-function');
+
+  assertNotNull('Function exists',
+      Blockly.mainBlockSpace.findFunction('test-usercreated-function'));
+  assertEquals('Delete button says "Delete"', 'Delete',
```

Comment:
probably want to throw in a "Function editor has delete button" test before this; if something broke and the delete button doesn't exist, this test will probably raise something like a `TypeError: Cannot read property 'textContent' of undefined` error, and the helper message will be confusing.